File: Basics/Palindrome/palind.go
package main

import (
	"fmt"
	"strings"
	"unicode"
)

func isPalindrome(s string) bool {
	s = strings.Map(func(r rune) rune {
		if unicode.IsLetter(r) || unicode.IsNumber(r) {
			return unicode.ToLower(r)
		}
		return -1
	}, s)
	
	for i, j := 0, len(s)-1; i < j; i, j = i+1, j-1 {
		if s[i] != s[j] {
			return false
		}
	}
	return true
}

func main() {
	stringToCheck := "A man, a plan, a canal: Panama"
	result := isPalindrome(stringToCheck)
	fmt.Println(result)
}

File: Basics/Palindrome/palind.rs
fn is_palindrome(s: &str) -> bool {
    let s: String = s.chars().filter(|c| c.is_alphanumeric()).map(|c| c.to_ascii_lowercase()).collect();
    let rev_s: String = s.chars().rev().collect();
    s == rev_s
}

fn main() {
    let string_to_check = "A man, a plan, a canal: Panama";
    let result = is_palindrome(string_to_check);
    println!("{}", result);
}
